weave/panels/panel_plot.py

```diff
@@ -302,7 +316,88 @@ class PlotConfig:
     legendSettings: typing.Optional[LegendSettings]
     configOptionsExpanded: typing.Optional[ConfigOptionsExpanded]
     signals: Signals
-    configVersion: int = 11
+    configVersion: int = 12
+
+
+def set_through_array(
+    obj: typing.Any, path: typing.List[str], value: typing.Any
+) -> None:
+    if len(path) == 0:
+        return
+
+    first, *rest = path
+
+    if first == "#":
+        if isinstance(obj, list):
+            for i, item in enumerate(obj):
+                set_through_array(
+                    item, rest, (value[i] if isinstance(value, list) else value)
+                )
+    elif len(rest) == 0:
+        if isinstance(obj, dict):
+            obj[first] = value
+        else:
+            setattr(obj, first, value)
+    elif hasattr(obj, first):
+        set_through_array(getattr(obj, first), rest, value)
+    else:
+        raise ValueError(f"Could not set {path} in {obj} to {value}")
+
+
+def get_through_array(
+    obj: typing.Any, path: typing.List[str], coerce_list_of_nodes_to_node=False
+) -> typing.Any:
+    if len(path) == 0:
+        return obj
+
+    first, *rest = path
+
+    if first == "#":
+        if isinstance(obj, list):
+            result = [
+                get_through_array(
+                    item,
+                    rest,
+                    coerce_list_of_nodes_to_node=coerce_list_of_nodes_to_node,
+                )
+                for item in obj
+            ]
+            if coerce_list_of_nodes_to_node and all(
+                isinstance(item, graph.Node) for item in result
+            ):
+                result = list_.make_list({i: item for i, item in enumerate(result)})
+            return result
+        else:
+            raise ValueError(f"Could not get {path} in {obj}: not a list")
+    elif hasattr(obj, first):
+        return get_through_array(
+            getattr(obj, first),
+            rest,
+            coerce_list_of_nodes_to_node=coerce_list_of_nodes_to_node,
+        )
+    elif len(rest) == 0:
+        if isinstance(obj, dict):
+            return obj[first]
+        else:
+            return getattr(obj, first, None)
+    else:
+        raise ValueError(f"Could not get {path} in {obj}")
+
+
+def ensure_node(
```

Comment:
Can you describe what's up with the set_through_array get_through_array stuff? Probably warrants a big comment block here.

Reply:
done